pkg/plugins/golang/deploy-image/v1alpha1/scaffolds/internal/templates/api/types.go

```diff
@@ -89,6 +89,8 @@ type {{ .Resource.Kind }}Spec struct {
 type {{ .Resource.Kind }}Status struct {
 	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
+
+	DeploymentStatus string ` + "`" + `json:"deploymentStatus,omitempty"` + "`" + `
```

Comment:
To follow the good practices and shows how to do it we should use conditional status therefore here we would have: (see the doc: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties)

```suggestion
	// Represents the observations of a {{ .Resource.Kind }}'s current state.
	// .{{ .Resource.Kind }}.status.conditions.type are: "Available", "Progressing", and "Degraded"
	// .{{ .Resource.Kind }}.status.conditions.status are one of True, False, Unknown.
	// .{{ .Resource.Kind }}.status.conditions.reason the value should be a CamelCase string and producers of specific
	// condition types may define expected values and meanings for this field, and whether the values
	// are considered a guaranteed API.
	// .{{ .Resource.Kind }}.status.conditions.Message is a human readable message indicating details about the transition.
	// For further information see: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties

	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,1,rep,name=conditions"`
```

Also, we can remove the comment since we are scaffolding the code.